agent.js

```diff
@@ -59,15 +59,37 @@ function checkNoProxy (uri) {
   return false
 }
 
+function getProcessEnv (env) {
```

Comment:
Can you export this function and write a small unit test for it? I think this is the kind of size that it's much easier to unit test :)